gcp_variant_transforms/libs/annotation/vep/vep_runner.py

```diff
@@ -0,0 +1,464 @@
+    return filesystems.FileSystems.join(dir_path, '')
+
+  def _get_output_log_path(self, output_dir, vm_index):
+    # type: (str, int) -> str
+    return '{}/logs/output_VM_{}'.format(output_dir, vm_index)
+
+  def _create_actions(self, input_file, output_file):
+    # type: (str, str) -> List
+    local_input_file = '/mnt/vep/{}'.format(_get_base_name(input_file))
+    return [
+        self._make_action('gsutil', '-q', 'cp', input_file, local_input_file),
+        self._make_action('rm', '-r', '-f', _LOCAL_OUTPUT_DIR),
+        self._make_action(_VEP_RUN_SCRIPT, local_input_file,
+                          _LOCAL_OUTPUT_FILE),
+        # TODO(bashir2): When the output files are local, the output directory
+        # structure should be created as well otherwise gsutil fails.
+        self._make_action('gsutil', '-q', 'cp', _LOCAL_OUTPUT_FILE,
+                          output_file)]
+
+
+class _SingleWorkerActions(object):
+  """Holds information about actions on a single virtual machine.
+
+  This is a pure data object and atributes can be accessed directly.
+  """
+
+  def __init__(self):
+    self.disk_size_bytes = 0
+    # `io_map` is a map from an input file to its corresponding output file.
+    self.io_map = {}  # type: Dict[str, str]
+
+  def __repr__(self):
+    return 'disk_size_bytes= {} , io_map= {}'.format(
+        self.disk_size_bytes, str(self.io_map))
+
+
+# TODO decide whether we want to unit-test this class in isolation and if so
+# move this and _SingleWorkerActions to a separate module.
+class _PipelinesExecutionInfo(object):
+  """This class determines actions to be done for running VEP on a set of files.
+
+  It also calculates some other configuration data for virtual machines running
+  these actions, like disk space.
+  """
+
+  # This is to account for the output file size plus some increate due to
+  # annotation fields.
+  _SIZE_FACTOR = 3
+
+  # The part of a path separating the file system scheme from the actual path.
+  _FILE_SYSTEM_SCHEME_SEPARATOR = '://'
+
+  # The expected suffix for .gz and .bgz files.
+  _GZ_SUFFIX = 'gz'
+
+  # This is used as a heuristic to account for the size of the unzipped file
+  # based on some anecdotal samples.
+  # TODO(bashir2): Revisit the file size calculation logic.
+  _GZ_FACTOR = 10
+
+  def __init__(self, file_metadata_list, output_dir, num_workers):
+    # type: (List[filesystem.FileMetadata], str, int) -> None
+    """
+    Args:
+      file_metadata_list: Information about input files, e.g., path, size, etc.
+      output_dir: The location of output files.
+      num_workers: Maximum number of workers to use.
+    """
+    self._single_vm_actions_list = []  # type: List[_SingleWorkerActions]
+    self._is_zip_input = False
+    self._output_dir = output_dir
+    self._num_workers = num_workers
+    self._input_size = len(file_metadata_list)
+
+    current_worker = _SingleWorkerActions()
+    for metadata in file_metadata_list:
```

Comment:
I think this logic can be simplified a lot. Essentially, you are dividing n tasks among m workers. I believe something like this should work:
```
file_metadata_chunks = [file_metadata_list[i::num_workers] for i in range(num_workers)]
for file_metadata_chunk in file_metadata_chunks:
  if not file_metadata_chunk: continue  # in case num files < num workers
  worker = _SingleWorkerActions()
  for file_metadata in file_metadata_chunk:
    worker.io_map[file_metadata.path] = ...
    worker.disk_size += ...
  self._single_vm_actions_list.append(worker)
```